### DISCUSSION TITLE
Easy way to use proper generation resolutions

### DISCUSSION THREAD

**Discussion by p-wegner** [Acly/krita-ai-diffusion#1254] [Ideas]
I'm not sure if this has been discussed yet, but although i really love the plugin for inpainting, it seems to be missing an essential and hopefully easy to implement feature. Instead (or additional) to setting a Maximum Pixel Count and a Resolution Multiplier hidden in the settings 
![image](https://github.com/user-attachments/assets/e922a111-daed-4883-9add-10968dd896ad), there could be an easy to reach option in the krita docker to set a fixed pixel count. 
E.g. for sd15 you would usually prefer resolutions between 512x512 pixels (best concept knowledge, anatomy, ...) up to 1024x1024 (with low denoise upscaling for fine details). Most sd15 finetunes handle 640x640 pixels just fine without too many distortions, but in general higher resolutions also reduce variance in the output while increasing quality to a degree (the knowledge seems to get lost).

With the current state, if i want to inpaint an area that is smaller than, lets say 300x300 pixels, i could 
- either upscale the whole image, inpaint and downscale manually
- set the Resolution multiplier in the Performance settings to 1.5x (highest possible slider value), although something between 1.7-2x would lead to optimal total pixel counts. With a factor of 1.5x, i would get a resolution of 450x450px. 

For inpainting large areas the same issue holds with downscaling. E.g. inpaint an area that is 2500x2500px with sd15 and your best bet is to set the multiplier to the lowest value i.e. 0.3x and you get a a 750x750 px generation that might not be optimal, but still better than any other value. 

**My proposed solution** is to do it similar to how [https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin](https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin) and [https://github.com/Bing-su/adetailer](https://github.com/Bing-su/adetailer) solve this issue.  Here you can define the actual resolution in pixels with two sliders. From a UX perspective this is overkill, because setting the total pixel count is usually sufficient. For sd15 reasonable values would be between 0.26 MP (512x512 px) and 1 MP and sdxl/flux might want something as high as 2-4MP

**Comment by Acly** [comment]
Actually this is so essential that it has been implemented since first release. You don't have to do anything manually.

You can tweak the "native checkpoint resolution" in Style > Advanced checkpoint settings if you want (default depends on model architecture), but there's usually no need.
